src/Services/M3U.vala

```diff
@@ -0,0 +1,72 @@
+/*
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ * SPDX-FileCopyrightText: 2025 elementary, Inc. (https://elementary.io)
+ */
+
+namespace Music.M3U {
+    public bool is_playlist (File suspicious_file) {
+        bool if_playlist = false;
+        try {
+            var info = suspicious_file.query_info (GLib.FileAttribute.STANDARD_CONTENT_TYPE, FileQueryInfoFlags.NONE);
+            var mimetype = info.get_content_type () ?? "";
+            if_playlist = mimetype == "audio/x-mpegurl";
+
+        } catch (Error e) {
+            warning (e.message);
+        }
+
+        return if_playlist;
+    }
```

Comment:
- Local variables doesn't need to be too specific
- Early returns allow the code readable and remove unnecessary variable

```suggestion
    public bool is_playlist (File file) {
        try {
            var info = file.query_info (GLib.FileAttribute.STANDARD_CONTENT_TYPE, FileQueryInfoFlags.NONE);
        } catch (Error e) {
            warning (e.message);
            return false;
        }

        var mimetype = info.get_content_type ();
        if (mimetype == null) {
            warning ("Failed to get content type");
            return false;
        }

        return mimetype == "audio/x-mpegurl";
    }
```